src/charm.py

```diff
@@ -919,6 +880,47 @@ def _generate_prometheus_config(self) -> bool:
         logger.info("Pushed new configuration")
         return True
 
+    def _process_tls_config(self, job) -> Dict:
+        certs = {}
+        if (tls_config := job.get("tls_config", {})) or job.get("scheme") == "https":
+            # Certs are transferred over relation data and need to be written to files on disk.
+            # CA certificate to validate the server certificate with.
+
+            # If the scrape job has a TLS section but no "ca_file", then use ours, assuming
+            # prometheus and all scrape jobs are signed by the same CA.
+            if ca_file := tls_config.get("ca_file") or self.cert_handler.ca:
+                # TODO we shouldn't be passing CA certs over relation data, because that
+                #  reduces to self-signed certs. Both parties need to separately trust the CA
+                #  instead.
```

Comment:
I agree with this TODO comment. Why are we configuring CA here at all. Shouldn't we just be using our system certs?

Reply:
This is not new, I just extract the code to a new method and keep the comments.
We have an issue to use secrets.